operator/api/v1alpha2/keptnappversion_types.go

```diff
@@ -111,6 +111,16 @@ func (a KeptnAppVersionList) GetItems() []client.Object {
 	return b
 }
 
+func (a *KeptnAppVersionList) RemoveDeprecated() {
+	var b []KeptnAppVersion
+	for _, i := range a.Items {
+		if i.Status.Status != common.StateDeprecated {
+			b = append(b, i)
+		}
```

Comment:
would it be faster using the [slices.Delete()](https://pkg.go.dev/golang.org/x/exp/slices#Delete) function?

Reply:
not actually, the function will return a new slice after deletion and this new slice won't be considered in the for loop, therefore we might get into troubles, what we actually can do is initialize an array before the for loop with the length of the original slice, what will be more effective as golang will no need to initialize and adapt the size of the slice automatically